src/magnify/apps/core/api.py

```diff
@@ -234,3 +202,125 @@ def perform_create(self, serializer):
         RoomUser.objects.create(
             room=room, user=self.request.user, is_administrator=True
         )
+
+    @action(
+        methods=["post"],
+        detail=True,
+        url_path="users",
+        permission_classes=[permissions.IsAuthenticated],
+    )
+    # pylint: disable=unused-argument,invalid-name
+    def users(self, request, pk=None):
+        """Adds a user in a room."""
+        room = self.get_object()
+        user = User.objects.get(id=request.data.get("user"))
+        serializer = RoomUserSerializer(
+            data={
+                "room": room.id,
+                "user": user.id,
+                "is_administrator": request.data.get("is_administrator"),
+            },
+            context={"request": request},
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=201, headers=headers)
+
+    @action(
+        methods=["get", "put", "delete"],
+        detail=True,
+        url_path="users/(?P<user_pk>[^/.]+)",
+        permission_classes=[IsRoomAdministrator],
+    )
+    # pylint: disable=unused-argument,invalid-name
+    def user(self, request, user_pk, pk=None):
+        """Get, update, or delete a user of a room."""
+        room = self.get_object()
+        room_user = RoomUser.objects.get(room=room, user__pk=user_pk)
+
+        if request.method == "DELETE":
+            room_user.delete()
+            return Response(status=204)
+
+        if request.method == "GET":
+            serializer = RoomUserSerializer(room_user)
+            return Response(serializer.data)
+
+        if request.method == "PUT":
+            user = User.objects.get(id=user_pk)
+            serializer = RoomUserSerializer(
+                room_user,
+                data={
+                    "room": room.id,
+                    "user": user.id,
+                    "is_administrator": request.data.get("is_administrator"),
+                },
```

Comment:
In fact it seems you are getting the user by its pk only to get its id, and the room only to get its id. 
```suggestion
            serializer = RoomUserSerializer(
                room_user,
                data={
                    "room": pk,
                    "user": user_pk,
                    "is_administrator": request.data.get("is_administrator"),
                },
```

Reply:
For the user, I agree!
Four the room, too, but apparently we have to call `self.get_object()` for django to check for `has_object_permission`. And calling it directly will require us to get it anyway.

Reply:
Ok good point!